File: tools/email_examples/node/sendgrid_example.js
const sg = require('@sendgrid/mail');

if (!process.env.SG_API_KEY && !process.env.SENDGRID_API_KEY) {
  console.error('Set SG_API_KEY or SENDGRID_API_KEY env var');
  process.exit(1);
}

sg.setApiKey(process.env.SG_API_KEY || process.env.SENDGRID_API_KEY);

const from = process.env.FROM_EMAIL || 'no-reply@example.com';
const to = process.env.TO_EMAIL || 'jimyount1947@gmail.com';

async function send() {
  const msg = {
    to,
    from,
    subject: 'Test email (SendGrid @sendgrid/mail)',
    text: 'Hello from SendGrid Node example',
  };

  const res = await sg.send(msg);
  console.log('SendGrid response status:', res[0]?.statusCode);
}

send().catch((err) => { console.error(err); process.exit(1); });

File: tools/email_examples/python/sendgrid_example.py
import os
import requests
import json

API_KEY = os.environ.get('SENDGRID_API_KEY') or os.environ.get('SG_API_KEY')
if not API_KEY:
    raise SystemExit('Set SENDGRID_API_KEY or SG_API_KEY environment variable')

from_addr = os.environ.get('FROM_EMAIL', 'no-reply@example.com')
to_addr = os.environ.get('TO_EMAIL', 'jimyount1947@gmail.com')

url = 'https://api.sendgrid.com/v3/mail/send'
payload = {
  "personalizations": [{"to":[{"email": to_addr}], "subject":"Test email (SendGrid API)"}],
  "from": {"email": from_addr},
  "content":[{"type":"text/plain","value":"Hello from SendGrid API (Python)"}]
}
headers = {'Authorization': f'Bearer {API_KEY}', 'Content-Type': 'application/json'}

r = requests.post(url, headers=headers, data=json.dumps(payload))
print(r.status_code, r.text)
